### PR TITLE
Feature/create notification test

### PR DESCRIPTION
**Pull request by victoria-kit**
## JIRA

* https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=933&view=detail&selectedIssue=PYTHON-214
* https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=933&view=detail&selectedIssue=PYTHON-213

## Code reviewers

- [ ] @meelros
- [ ] @BoroviyOrest
- [ ] @Kardan1123
- [ ] @dimigor
- [ ] @PetrushynskyiOleksii
- [x] @Sasha1152

### Second Level Review

- [ ] @lhalam

## Summary of change

ToDo
Modified notification validation, 
Create tests for Notification model, 
Create tests for Notification views 
Replace responses in Notification View to constants

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by PetrushynskyiOleksii — CHANGES_REQUESTED** [review]
You also should test the next lines of code::

- in `PUT`:

    ```
    if not way == notification.way: 
      return RESPONSE_403_ACCESS_DENIED
    ```
- in `POST`:

    ```
    if not user == way.user:
        return RESPONSE_403_ACCESS_DENIED
    ```
- in `DELETE`
    ```
     if not way == notification.way:
        return RESPONSE_403_ACCESS_DENIED
   ```

File: way_to_home/notification/views.py [hunk 1]
```diff
@@ -50,18 +59,18 @@ def put(self, request, way_id, notification_id=None):  # pylint: disable=R0201,
         data = request.body
         way = Way.get_by_id(obj_id=way_id)
 
-        if not (way or notification_id):
-            return HttpResponse('failed, objects not found', status=400)
+        if not (way and notification_id):
```
Comment (dimigor) [inline way_to_home/notification/views.py]: Must be `or`.
In your case this return handles when two variables are False, but we need when even one is False

File: way_to_home/notification/views.py [hunk 2]
```diff
@@ -113,21 +121,21 @@ def delete(self, request, way_id, notification_id=None):  # pylint: disable=R020
         user = request.user
         way = Way.get_by_id(obj_id=way_id)
 
-        if not (way or notification_id):
-            return HttpResponse('failed, objects not found', status=400)
+        if not (way and notification_id):
```
Comment (dimigor) [inline way_to_home/notification/views.py]: Same as above

File: way_to_home/tests/unittests/notification/testviews.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+from notification.models import Notification
+from way.models import Way
+
+
+class NotificationViewsTestCase(TestCase):
+    """TestCase for providing Notification views testing."""
+
+    def setUp(self):
+        """Method that provides preparation before testing Notification views."""
+        user = CustomUser(id=100, email='testuser@mail.com', is_active=True)
```
Comment (dimigor) [inline way_to_home/tests/unittests/notification/testviews.py]: use `CustomUser.objects.create`

File: way_to_home/tests/unittests/notification/testviews.py [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+    def test_post_empty_json(self):
+        """Method that tests unsuccessful post request with empty JSON data."""
+
+        data = {}
+        url = reverse('notification',
+                      kwargs={'way_id': self.notification.way_id, 'notification_id': self.notification.id})
+        response = self.client.post(url, json.dumps(data, cls=DjangoJSONEncoder), content_type='application/json')
+        self.assertEqual(response.status_code, 400)
+
+    def test_post_from_another_way(self):
```
Comment (dimigor) [inline way_to_home/tests/unittests/notification/testviews.py]: I think this method should be named as "test_post_from_non_existing_way"
Also, add test when way exist, but the user is not the owner, and check if `response.status_code` equals to 403

Comment (victoria-kit) [reply way_to_home/tests/unittests/notification/testviews.py]: done, thank's

File: way_to_home/tests/unittests/notification/testviews.py (hunk 3 above)
Comment (victoria-kit) [reply way_to_home/tests/unittests/notification/testviews.py]: With CustomUser.objects.create user can't login.

File: way_to_home/notification/views.py (hunk 2 above)
Comment (victoria-kit) [reply way_to_home/notification/views.py]: done. I change to "if not way or not notification_id:"

File: way_to_home/notification/views.py (hunk 1 above)
Comment (victoria-kit) [reply way_to_home/notification/views.py]: done

File: way_to_home/tests/unittests/notification/testviews.py (hunk 3 above)
Comment (victoria-kit) [reply way_to_home/tests/unittests/notification/testviews.py]: done

**Review by dimigor — DISMISSED** [review]

**Review by PetrushynskyiOleksii — DISMISSED** [review]

**Review by dimigor — APPROVED** [review]

**Review by meelros — APPROVED** [review]
All is good!

**Review by Sasha1152 — APPROVED** [review]

**Pull request merged**